unic/char/range/src/range.rs

```diff
@@ -0,0 +1,172 @@
+use std::char;
+use std::collections::Bound;
+use CharIter;
+
+/// A range of unicode code points.
+///
+/// The most idiomatic way to construct this range is through the use of the `chars!` macro:
+///
+/// ```
+/// # #[macro_use] extern crate unic_char_range;
+/// # use unic_char_range::*;
+/// # fn main() {
+/// assert_eq!(chars!('a'..='z'), CharRange::closed('a', 'z'));
+/// assert_eq!(chars!('a'..'z'), CharRange::open_right('a', 'z'));
+/// assert_eq!(chars!(..), CharRange::all());
+/// # }
+/// ```
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct CharRange {
+    /// The lowest character in this range (inclusive).
+    pub low: char,
+
+    /// The highest character in this range (inclusive).
+    pub high: char,
+}
+
+/// Constructors
+impl CharRange {
+    /// Construct a closed range of characters.
+    ///
+    /// If `stop` is ordered before `start`, the resulting range will be empty.
```

Comment:
I think this line can be moved to the type doc, with addition of `eq()` behavior. (And we can drop it from constructors.) (From d11n perspective, since the fields are `pub`, there shouldn't be a need to read the constructor docs to understand  the interpretation of the fields.)